lib/ws-router.js

```diff
@@ -0,0 +1,37 @@
+module.exports = WebsocketRouter;
+
+var assert = require('assert');
+var WebsocketChannel = require('./ws-channel');
+
+
+function WebsocketRouter(app, path) {
+  if (!(this instanceof WebsocketRouter))
+    return new WebsocketRouter(app, path);
+
+  this._channels = {};
+  this._path = path;
+  app.ws(path, this.handleSocket.bind(this));
+}
+
+
+WebsocketRouter.prototype.createChannel =
+    function createChannel(onRequest, onNotification) {
+  var channel = WebsocketChannel.create(onRequest, onNotification);
+  var token = channel.createToken();
```

Comment:
This works fine for pm with the direct driver, because all supervisor's are direct children, started after `createToken()` is called.

It won't work so well for docker, in which case pm may be restarting, but there is an existing docker container running, that already has a token, and it's supervisor will be using it to try to connect to us.

The same issue comes up for executor, which will need to be allocated a token at time of provisioning by making a REST API call to central. central will persist that token in the Executor model, and when central is restarted, it will create channels for all executors that are registered.

I suggest createChannel get a final optional argument, `token`, and if its null generate it, if its non-null, use it (but assert that it is unique).

Also, to delete executors, central will need a `removeChannel(token)` API.

^--- none of above should block merging this PR... this just seemed a good place to comment.

/to @kraman  I notice that the Executor model in strong-mesh-models isn't up to date with new mesh architecture. The address maybe can stay, and be dynamic (noticed at time executor connects). Its purely for debugging/listing it via REST. The hostname is probably unneeded, and unlikely to be useful. The APIPort doesn't exist anymore, since Executor won't offer a REST API. ... And it needs a Token String to persist, to use as above.